src/main/scala/inox/solvers/z3/Z3Native.scala

```diff
@@ -654,7 +654,26 @@ trait Z3Native extends ADTManagers with Interruptible { self: AbstractSolver =>
                     FiniteSet(elems.toSeq, dt)
                 }
 
-              case StringType() => StringLiteral(utils.StringUtils.decode(z3.getString(t)))
+              case StringType() if z3.getDeclKind(decl) == OpSeqEmpty =>
+                assert(args.isEmpty)
+                StringLiteral("")
+
+              case StringType() if z3.getDeclKind(decl) == OpSeqUnit =>
+                assert(args.size == 1)
+                rec(args.head, BVType(false, 16), seen) match {
+                  case lit@BVLiteral(_, _, _) =>
+                    StringLiteral(String.valueOf(lit.toBigInt.toChar))
+                  case rarg => rarg
```

Comment:
What case are we expecting this to match? Won't we run into type issues by returning something with expected type `BVType(false, 16)` as a string?

Reply:
You are right, I did not think about that. I guess matching that `rarg` branch will make a fine test case :p

Reply:
I have looked at how `Z3Target` handles that similar case, and it appears that it expects a (char) literal. It seems we cannot do much, so I've put a call to `unsound`.